include/aws/io/stream.h

```diff
@@ -39,6 +39,17 @@ struct aws_input_stream_vtable {
     int (*read)(struct aws_input_stream *stream, struct aws_byte_buf *dest);
     int (*get_status)(struct aws_input_stream *stream, struct aws_stream_status *status);
     int (*get_length)(struct aws_input_stream *stream, int64_t *out_length);
+
+    /**
+     * Optional. If not set, the default aws_ref_count_acquire/release will be used.
+     * Set for high level language binding that has its own refcounting implementation and needs to be kept alive from
+     * C.
+     * When the C level needs to keep the high level language binding alive, while binding as the user also holds the
+     * reference to the C object, it will result in a recursive reference.
+     * The function here is for C Object to use the high level refcount implementation directly.
+     */
+    struct aws_input_stream *(*acquire)(struct aws_input_stream *stream);
+    struct aws_input_stream *(*release)(struct aws_input_stream *stream);
```

Comment:
this doesn't need to return anything
you can do that from aws_input_stream_acquire()

Reply:
This is something I grumped about earlier and twisted Dengke's arm to include.  I prefer the return value of always-null to simplify non-final cleanup (just assign the result) and swaps

Reply:
yeah, we agreed that the c function should return the pointer, but the implementation doesn't need to.